src/etl/sars-cov-2/steps/add-vaccination-data-to-estimate-step.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  EstimateFieldsAfterJitteringPinLatLngStep,
+  StructuredPositiveCaseDataAfterJitteringPinLatLngStep,
+  StructuredVaccinationDataAfterJitteringPinLatLngStep,
+} from "./jitter-pin-lat-lng-step";
+
+export type EstimateFieldsAfterAddingVaccinationDataStep =
+  EstimateFieldsAfterJitteringPinLatLngStep & {
+    countryPeopleVaccinatedPerHundred: number | undefined;
+    countryPeopleFullyVaccinatedPerHundred: number | undefined;
+  };
+export type StructuredVaccinationDataAfterAddingVaccinationDataStep =
+  StructuredVaccinationDataAfterJitteringPinLatLngStep;
+export type StructuredPositiveCaseDataAfterAddingVaccinationDataStep =
+  StructuredPositiveCaseDataAfterJitteringPinLatLngStep;
+
+interface AddVaccinationDataToEstimateStepInput {
+  allEstimates: EstimateFieldsAfterJitteringPinLatLngStep[];
+  vaccinationData: StructuredVaccinationDataAfterJitteringPinLatLngStep;
+  positiveCaseData: StructuredPositiveCaseDataAfterJitteringPinLatLngStep;
+}
+
+interface AddVaccinationDataToEstimateStepOutput {
+  allEstimates: EstimateFieldsAfterAddingVaccinationDataStep[];
+  vaccinationData: StructuredVaccinationDataAfterAddingVaccinationDataStep;
+  positiveCaseData: StructuredPositiveCaseDataAfterAddingVaccinationDataStep;
+}
+
+export const addVaccinationDataToEstimateStep = (
+  input: AddVaccinationDataToEstimateStepInput
+): AddVaccinationDataToEstimateStepOutput => {
+  console.log(
+    `Running step: addVaccinationDataToEstimateStep. Remaining estimates: ${input.allEstimates.length}`
+  );
+
+  return {
+    allEstimates: input.allEstimates.map((estimate) => {
+      const {
+        countryPeopleVaccinatedPerHundred,
+        countryPeopleFullyVaccinatedPerHundred
+      } = input.vaccinationData
+        .find((element) => element.threeLetterCountryCode === estimate.countryAlphaThreeCode)?.data
+        .find((element) => estimate.samplingMidDate && element.year === estimate.samplingMidDate.getUTCFullYear().toString())?.data
+        .find((element) => estimate.samplingMidDate && element.month === (estimate.samplingMidDate.getUTCMonth() + 1).toString())?.data
+        .find((element) => estimate.samplingMidDate && element.day === (estimate.samplingMidDate.getUTCDate()).toString()) ?? {
+          countryPeopleVaccinatedPerHundred: undefined,
+          countryPeopleFullyVaccinatedPerHundred: undefined
+        }
```

Comment:
I'm aware the data is incomplete on certain timeframes some of these fields are only available on specific days. I'm thinking of something to make this better just let it :cook: for a sec :)